avaframe/com1DFA/particleTools.py

```diff
@@ -0,0 +1,640 @@
+    # note that if we did not update the particles FD, we use here the h from the previous time step
+    aPart = mPart/(rho*hPart)
+    rNew = distSplitPart * np.sqrt(aPart/math.pi)
+    cos = rNew*np.array([-1, 1])
+    # compute velocity mag to get the direction of the flow (e_1)
+    uMag = DFAtls.norm(uxPart, uyPart, uzPart)
+    xNew = xPart + cos * uxPart/uMag
+    yNew = yPart + cos * uyPart/uMag
+    zNew = zPart + cos * uzPart/uMag
+    # toDo: do we need to reproject the particles on the dem?
+    return xNew, yNew, zNew
+
+
+def getTangenVectors(nx, ny, nz, ux, uy, uz):
+    """Compute the tangent vector to the surface
+
+    If possible, e1 is in the velocity direction, if not possible,
+    use the tangent vector in x direction for e1 (not that any other u vector could be provided,
+    it does not need to be the velocity vector, it only needs to be in the tangent plane)
+    Parameters
+    ----------
+    nx : float
+        x component of the normal vector
+    ny : float
+        y component of the normal vector
+    nz : float
+        z component of the normal vector
+    ux : float
+        x component of the velocity vector
+    uy : float
+        y component of the velocity vector
+    uz : float
+        z component of the velocity vector
+
+    Returns
+    -------
+    e1x : float
+        x component of the first tangent vector
+    e1y : float
+        y component of the first tangent vector
+    e1z : float
+        z component of the first tangent vector
+    e2x : float
+        x component of the second tangent vector
+    e2y : float
+        y component of the second tangent vector
+    e2z : float
+        z component of the second tangent vector
+    """
+    # compute the velocity magnitude
+    velMag = DFAtls.norm(ux, uy, uz)
+    if velMag > 0:
+        e1x = ux / velMag
+        e1y = uy / velMag
+        e1z = uz / velMag
+    else:
+        # if vector u is zero use the tangent vector in x direction for e1
+        e1x = np.array([1])
+        e1y = np.array([0])
+        e1z = -nx/nz
+        e1x, e1y, e1z = DFAtls.normalize(e1x, e1y, e1z)
+    # compute the othe tengent vector
+    e2x, e2y, e2z = DFAtls.crossProd(nx, ny, nz, e1x, e1y, e1z)
+    e2x, e2y, e2z = DFAtls.normalize(e2x, e2y, e2z)
+    return e1x, e1y, e1z, e2x, e2y, e2z
+
+
+def findParticles2Track(particles, center, radius):
+    '''Find particles within a circle arround a given point
+
+    Parameters
+    ----------
+    particles : dict
+        particles dictionary (with the 'parentID' array)
+    center : dict
+        point dictionary:
+            x : x coordinate
+            y : y coordinate
+            z : z coordinate
+    radius : float
+        radius of the circle around point
+
+    Returns
+    -------
+    particles2Track : numpy array
+        array with Parent ID of particles to track
+    track: boolean
+        False if no particles are tracked
+    '''
+    track = True
```

Comment:
test flag? or move to ini, or what is it needed for?

Reply:
this variable might be modified by this function, the track is then returned